src/widgetastic/widget/table.py

```diff
@@ -0,0 +1,1055 @@
+                    self.rows_ignore_bottom > 0):
+                rows = rows[:-self.rows_ignore_bottom]
+            row_count = len(rows)
+            present_row_values = value[:row_count]
+            if total_values > row_count:
+                extra_row_values = value[row_count:]
+            else:
+                extra_row_values = []
+            changed = any(row.fill(value) for row, value in zip(rows, present_row_values))
+            for extra_value in extra_row_values:
+                if self[self.row_add()].fill(extra_value):
+                    changed = True
+            return changed
+
+    @property
+    def row_count(self):
+        """Returns how many rows are currently in the table."""
+        return len(self.browser.elements(self.ROWS, parent=self))
+
+    def row_add(self):
+        """To be implemented if the table has dynamic rows.
+
+        This method is called when adding a new row is necessary.
+
+        Default implementation shouts :py:class:`NotImplementedError`.
+
+        Returns:
+            An index (position) of the new row. ``None`` in case of error.
+        """
+        raise NotImplementedError(
+            'You need to implement the row_add in order to use dynamic adding')
+
+    def row_save(self, row=None):
+        """To be implemented if the table has dynamic rows.
+
+        Used when the table needs confirming saving of each row.
+
+        Default implementation just writes a debug message that it is not used.
+        """
+        self.logger.debug('Row saving not used.')
+
+    @property
+    def has_rowcolspan(self):
+        """Checks whether table has rowspan/colspan attributes"""
+        return bool(self.browser.elements('.//td[@rowspan or @colspan]', parent=self))
+
+    def _process_table(self):
+        queue = deque()
+        tree = Node(name=self.browser.tag(self), obj=self, position=None)
+        queue.append(tree)
+
+        while len(queue) > 0:
+            node = queue.popleft()
+            # visit node
+            children = self.browser.elements('./*[descendant-or-self::node()]', parent=node.obj)
+            for position, child in enumerate(children):
+                cur_tag = self.browser.tag(child)
+                if cur_tag == 'tr':
+                    # todo: add logger
+                    # position has been decremented for rows because it is incremented
+                    # if 0 in TableRow.__init__ for some reason
+                    cur_obj = TableRow(parent=self._get_ancestor_node_obj(node), index=position)
+                    cur_node = Node(name=cur_tag, parent=node, obj=cur_obj, position=position)
+                    queue.append(cur_node)
+                elif cur_tag == 'td':
+                    cur_position = self._get_position_respecting_spans(node)
+                    cur_obj = TableColumn(parent=node.obj, position=cur_position)
+                    Node(name=cur_tag, parent=node, obj=cur_obj, position=cur_position)
```

Comment:
I think you mean to assign this Node instance to `cur_node`? Otherwise what's the purpose?

Reply:
assignment is done in root node. the rest of nodes is tied to parent nodes thru parent attribute